contracts/core/mpl-rewards/v1/test/MplRewards.t.sol

```diff
@@ -1,157 +1,483 @@
+    }
+
+    function mplRewards_recoverERC20(MplRewards mplRewards, address tokenAddress, uint256 amount) external {
+        mplRewards.recoverERC20(tokenAddress, amount);
+    }
+
+    function try_mplRewards_recoverERC20(address mplRewards, address tokenAddress, uint256 amount) external returns (bool ok) {
+        (ok,) = mplRewards.call(abi.encodeWithSignature("recoverERC20(address,uint256)", tokenAddress, amount));
+    }
+
+    function mplRewards_setRewardsDuration(MplRewards mplRewards, uint256 duration) external {
+        mplRewards.setRewardsDuration(duration);
+    }
+
+    function try_mplRewards_setRewardsDuration(address mplRewards, uint256 duration) external returns (bool ok) {
+        (ok,) = mplRewards.call(abi.encodeWithSignature("setRewardsDuration(uint256)", duration));
+    }
+
+    function mplRewards_setPaused(MplRewards mplRewards, bool paused) external {
+        mplRewards.setPaused(paused);
+    }
+
+    function try_mplRewards_setPaused(address mplRewards, bool paused) external returns (bool ok) {
+        (ok,) = mplRewards.call(abi.encodeWithSignature("setPaused(bool)", paused));
+    }
+
+    function mplRewards_stake(MplRewards mplRewards, uint256 amount) external {
+        mplRewards.stake(amount);
     }
 
-    /*******************************/
-    /*** Admin Functions Testing ***/
-    /*******************************/
-    function test_transferOwnership() public {
-        assertEq(mplRewards.owner(), address(gov));
+    function try_mplRewards_stake(address mplRewards, uint256 amount) external returns (bool ok) {
+        (ok,) = mplRewards.call(abi.encodeWithSignature("stake(uint256)", amount));
+    }
+
+    function mplRewards_withdraw(MplRewards mplRewards, uint256 amount) external {
+        mplRewards.withdraw(amount);
+    }
+
+    function try_mplRewards_withdraw(address mplRewards, uint256 amount) external returns (bool ok) {
+        (ok,) = mplRewards.call(abi.encodeWithSignature("withdraw(uint256)", amount));
+    }
 
-        assertTrue(!fakeGov.try_transferOwnership(address(fakeGov)));
-        assertTrue(     gov.try_transferOwnership(address(fakeGov)));
+    function mplRewards_getReward(MplRewards mplRewards) external {
+        mplRewards.getReward();
+    }
 
-        assertEq(mplRewards.owner(), address(fakeGov));
+    function try_mplRewards_getReward(address mplRewards) external returns (bool ok) {
+        (ok,) = mplRewards.call(abi.encodeWithSignature("getReward()"));
+    }
 
-        assertTrue(   !gov.try_transferOwnership(address(gov)));
-        assertTrue(fakeGov.try_transferOwnership(address(gov)));
+    function mplRewards_exit(MplRewards mplRewards) external {
+        mplRewards.exit();
+    }
 
-        assertEq(mplRewards.owner(), address(gov));
+    function try_mplRewards_exit(address mplRewards) external returns (bool ok) {
+        (ok,) = mplRewards.call(abi.encodeWithSignature("exit()"));
     }
```

Comment:
All of these functions exist in the `/users` directory, we should just import and instantiate the users in the same way. I agree with the changes around mocking the contracts and making them generic, but I think we should maintain the same users structure that we have from the previous test.

Reply:
I disagree with a global users directory because it results in the possibility for more than one unrelated contract/test using the same user, so making changes to the user to satisfy one could potentially break something for the other. All the users in the global users directory come with a lot of baggage and aren't functional, and I would rather not have a user that is half functional and half not functional. Eventually, when all the user's are all functional, we can merge them back into some global sers directory perhaps.